examples/psyad/lfric/tangent_linear_tweaked/tl_moist_dyn_gas_kernel_mod_tweaked.F90

```diff
@@ -0,0 +1,86 @@
+module tl_moist_dyn_gas_kernel_mod
+
+    use argument_mod,                  only: arg_type,          &
+                                             GH_FIELD, GH_REAL, &
+                                             GH_SCALAR,         &
+                                             GH_WRITE, GH_READ, &
+                                             CELL_COLUMN
+    use constants_mod,                 only: r_def, i_def
+    use fs_continuity_mod,             only: Wtheta
+    use kernel_mod,                    only: kernel_type
+    use planet_config_mod,             only: recip_epsilon
+
+    implicit none
+
+    private
+
+    !--------------------------------------------------------------------------
+    ! Public types
+    !--------------------------------------------------------------------------
+    !> The type declaration for the kernel. Contains the metadata needed by the Psy layer
+    type, public, extends(kernel_type) :: tl_moist_dyn_gas_kernel_type
+        private
+        type(arg_type) :: meta_args(2) = (/                 &
+             arg_type(GH_FIELD, GH_REAL, GH_WRITE, Wtheta), &
+             arg_type(GH_FIELD, GH_REAL, GH_READ,  Wtheta)  &
+             /)
+        integer :: operates_on = CELL_COLUMN
+    contains
+        procedure, nopass :: tl_moist_dyn_gas_code
+    end type
+
+    !--------------------------------------------------------------------------
+    ! Contained functions/subroutines
+    !--------------------------------------------------------------------------
+    public :: tl_moist_dyn_gas_code
+
+contains
+
+    !> @brief Compute the tangent linear moist dynamical factors
+    !! @param[in]     nlayers       Integer the number of layers
+    !! @param[in,out] moist_dyn_gas Change in Gas factor (m_v / epsilon)
+    !! @param[in]     mr_v          Change in Water vapour mixing ratio
+    !! @param[in]     ndf_wtheta    The number of degrees of freedom per cell
+    !!                              for wtheta
+    !! @param[in]     udf_wtheta    The number of total degrees of freedom for
+    !!                              wtheta
+    !! @param[in]     map_wtheta    Integer array holding the dofmap for the
+    !!                              cell at the base of the column
+    subroutine tl_moist_dyn_gas_code( nlayers, moist_dyn_gas, mr_v,         &
+                                      ndf_wtheta, undf_wtheta, map_wtheta )
+
+        implicit none
+
+        ! Arguments
+        integer(kind=i_def), intent(in) :: nlayers, ndf_wtheta, undf_wtheta
+
+        real(kind=r_def), dimension(undf_wtheta),   intent(inout) :: moist_dyn_gas
+        real(kind=r_def), dimension(undf_wtheta),   intent(in)    :: mr_v
+        integer(kind=i_def), dimension(ndf_wtheta), intent(in)    :: map_wtheta
+
+        ! Internal variables
+        integer(kind=i_def)                 :: k, df
+
+        real(kind=r_def)                    :: mr_v_at_dof
+
+        real(kind=r_def)                    :: recip_epsilon_tmp
```

Comment:
Worth a comment here saying that this local var needs to be removed in favour of the imported var once the
adjoint is created? (Why is this a problem for PSyAD?)

Reply:
I've added some words but what you are asking should really be replicated for all tweaks and I had kept away from this as I thought diff's were OK, as we will hopefully be fixing the issues. However, if we are expecting others to make manual changes and understand the limitations then I agree that comments should be added with reference to issues, but not just for this change, for all of the changes.

Reply:
I think at this stage of the game we'll just proceed as we are. I didn't understand the need for this particular tweak but now I do :-) Should be an easy limitation to fix.